### PR TITLE
(SERVER-2550) Add `crl-issued-by-cert?` function

### PR DESCRIPTION
**Pull request by mwaggett**
This commit adds a function to determine whether the given CRL was issued by
the given certificate. This will be used to verify CRLs against certificate
chains, particularly when consuming CRL updates from an external root.

### CODE REVIEW

**Comment by mwaggett** [comment]
I sorta felt like this didn't really need a test? but happy to add one if y'all think it's worthwhile.

File: src/clojure/puppetlabs/ssl_utils/core.clj [hunk 1]
```diff
@@ -579,13 +579,20 @@
...
 (schema/defn ^:always-validate pem->ca-crl :- X509CRL
   "Given a CRL chain and CA certificate, extract the CRL issued by the
   certificate"
   [crl-chain :- Readerable
    ca-cert :- X509Certificate]
-  (let [match-ca-cert? (fn [crl] (= (.getIssuerX500Principal crl)
-                                    (.getSubjectX500Principal ca-cert)))
+  (let [match-ca-cert? (fn [crl] (crl-issued-by-cert? crl ca-cert))
         crls (pem->crls crl-chain)
         crl (first (filter match-ca-cert? crls))]
```
Comment (Magisus) [inline src/clojure/puppetlabs/ssl_utils/core.clj:597]: Side note, this is some pretty good evidence that the order of the CRLs does not need to matter.

**Review by Magisus — APPROVED** [review]

File: src/clojure/puppetlabs/ssl_utils/core.clj (hunk 1 above)
Comment (mwaggett) [reply src/clojure/puppetlabs/ssl_utils/core.clj:597]: yeah, I saw that, but wasn't sure if there were other places we were making assumptions. like, maybe this lib doesn't care, but we care in puppetserver? idk.

Comment (Magisus) [reply src/clojure/puppetlabs/ssl_utils/core.clj:597]: Yeah. When you do the check, be sure to try revoking a cert, make sure that code path doesn't have any gotchas.

**Pull request merged**